src/main/java/com/zerobase/SelfWash/member/service/impl/OwnerSignUpServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.zerobase.SelfWash.member.service.impl;
+
+import com.zerobase.SelfWash.member.domain.entity.Owner;
+import com.zerobase.SelfWash.member.domain.form.SignUpForm;
+import com.zerobase.SelfWash.member.domain.repository.OwnerRepository;
+import com.zerobase.SelfWash.member.service.SignUpService;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OwnerSignUpServiceImpl implements SignUpService {
+
+  private final OwnerRepository ownerRepository;
+
+  @Override
+  public boolean support(String userType) {
+    return "owner".equalsIgnoreCase(userType);
+  }
+
+  @Override
+  public void signUp(SignUpForm form) {
+    Optional<Owner> optionalOwner = ownerRepository.findByEmail(form.getEmail());
+    if (optionalOwner.isPresent()) {
+      throw new RuntimeException("이미 존재하는 이메일입니다.");
+    }
+
+    ownerRepository.save(Owner.signUpFrom(form));
+  }
+
+  @Override
+  public void verifyEmail(String email, String key) {
+    Optional<Owner> optionalOwner = ownerRepository.findByEmail(email);
+    if (!optionalOwner.isPresent()) {
+      throw new RuntimeException("존재하지 않는 이메일입니다.");
+    }
+    Owner owner = optionalOwner.get();
```

Comment:
```suggestion
    Owner owner = ownerRepository.findByEmail(email).orElseThrow(() -> new RuntimeException("존재하지 않는 이메일입니다."));
```
Optional의 선언은 지양해주시면 좋겠습니다. 가능하면 chaining을 걸어서 Optional이 null인경우에 대한 처리도 꼭 같이 붙여주시면 좋을 것 같아요
https://dev-coco.tistory.com/178